modules/javafx.graphics/src/test/java/test/javafx/scene/PixelBufferTest.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright (c) 2019, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.javafx.scene;
+
+
+import java.nio.ByteBuffer;
+import java.nio.IntBuffer;
+
+import javafx.geometry.Rectangle2D;
+import javafx.scene.image.Image;
+import javafx.scene.image.PixelBuffer;
+import javafx.scene.image.PixelFormat;
+import javafx.scene.image.WritableImage;
+import javafx.util.Callback;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.fail;
+import org.junit.Test;
+
+public final class PixelBufferTest {
+    static final int WIDTH = 10;
+    static final int HEIGHT = 10;
+    static final PixelFormat<ByteBuffer> byteBGRAPrePf = PixelFormat.getByteBgraPreInstance();
+    static final ByteBuffer byteBuffer = ByteBuffer.allocateDirect(WIDTH * HEIGHT * 4);
+
+    static final PixelFormat<IntBuffer> intARGBPrePf = PixelFormat.getIntArgbPreInstance();
+    static final IntBuffer intBuffer = IntBuffer.allocate(WIDTH * HEIGHT);
+
+    @Test
+    public void testCreatePixelBufferByteBGRAPre() {
+        PixelBuffer pb = new PixelBuffer(WIDTH, HEIGHT, byteBuffer, byteBGRAPrePf);
+        assertEquals(WIDTH, pb.getWidth());
+        assertEquals(HEIGHT, pb.getHeight());
+        assertSame(byteBuffer, pb.getBuffer());
+        assertSame(byteBGRAPrePf, pb.getPixelFormat());
+    }
+
+    @Test
+    public void testCreatePixelBufferLessCapacityBuffer() {
+        try {
+            PixelBuffer pb = new PixelBuffer(WIDTH, HEIGHT + 1, byteBuffer, byteBGRAPrePf);
+            fail("Expected IllegalArgumentException");
+        } catch (IllegalArgumentException e) {
+        }
+    }
```

Comment:
I would add a test for the overflow case (where w*h is a negative number and the buffer is too small). It will fail until you make the change I suggested in the implementation. I also recommend repeating the buffer too small tests for the IntBuffer case.